demos/2025-07-30/slides/os-updates.md

```diff
@@ -0,0 +1,124 @@
+# Operating System A/B Updates
+
+### ...and delta updates
+
+---
+
+# What are OS A/B Updates
+
+* Full device update
+    * Linux kernel and device tree
+    * Operating System (root filesystem, aka root-fs)
+    * User applications
+
+* Two copies of the partitions are stored
+    * Apply update to the inactive partition
+    * Rollback to previous partition on an error
+
+---
+
+# What are OS A/B Updates (cont.)
+
+* OS Updates are generally large 😥
+    * Yocto images ~200MB (headless and with container engine)
+    * Debian based ~700MB (headless and with container engine)
+    * Debian with Desktop ~2GB
+
+---
+
+# Delta updates
+
+* Sending only "what changed" between two updates to reduce the download size
+
+---
+
+# Dynamic vs. Static Delta Updates
+
+* Method 1: Dynamic
+    * One file per release
+    * Index file is included as part of the update
+    * Device must download the relevant "parts" of the image
+    * More difficult to predict how "efficient" the download will be
+
+* Method 2: Static
+    * Delta between two specific versions is generated before sending to the device
+    * Delta file is typically smaller (download size is known upfront)
+    * More complex hosting as there are multiple files per release
+
+---
+
+# Dynamic Updates
+
+![dynamic-updates](dynamic-updates.drawio.png)
+
+---
+
+# Static Updates
+
+![static-updates](static-updates.drawio.png)
+
+---
+
+# Technologies that support delta updates
+
+|Tech.|Dynamic|Static|
+|-----|:-------:|:------:|
+|Rauc|✅|🚫|
+|mender|🚫|✅ (if you pay for it)|
+|Rugix|✅|✅|
+
+Note: SWUpdate might also support some of these features but wasn't part of the comparison
+
+---
+
+# What is Rugix?
+
+Rugix is an open-source tool suite to build reliable embedded Linux devices with efficient and secure over-the-air (OTA) update capabilities.
+
+* Rugix is actually now composed of two components
+    * Rugix (perform reliable A/B updates)
```

Comment:
I usually refer to this component as Rugix Ctrl now. I'd consider just calling that component Rugix but decided against it to avoid confusion. Do you think, it would be better to simply call it Rugix (without the Ctrl suffix)?